File: task3/Javascript/Subtask4.js
const fs = require('fs');

// Function to generate the diamond pattern
function generateDiamond(n) {
    let pattern = '';

    // Generate the upper part of the diamond
    for (let i = 0; i < n; i++) {
        pattern += ' '.repeat(n - i - 1) + '*'.repeat(2 * i + 1) + '\n';
    }

    // Generate the lower part of the diamond
    for (let i = n - 2; i >= 0; i--) {
        pattern += ' '.repeat(n - i - 1) + '*'.repeat(2 * i + 1) + '\n';
    }

    return pattern;
}

// Read the number from vsofficial.txt
fs.readFile('vsofficial.txt', 'utf8', (err, data) => {
    if (err) {
        console.error('Error reading the file:', err);
        return;
    }

    // Parse the number from the file
    const n = parseInt(data, 10);

    if (isNaN(n) || n <= 0) {
        console.error('Invalid number in vsofficial.txt. Please provide a valid positive number.');
        return;
    }

    // Generate the diamond pattern
    const pattern = generateDiamond(n);

    // Write the pattern to output.txt
    fs.writeFile('output.txt', pattern, (err) => {
        if (err) {
            console.error('Error writing to the file:', err);
        } else {
            console.log('Diamond pattern has been written to output.txt.');
        }
    });
});

File: task3/Java/Subtask4.java
import java.io.*;

public class Subtask4 {
    public static void main(String[] args) {
        String inputFileName = "vsofficial.txt";
        String outputFileName = "output.txt";
        int n;

        // Read the number from vsofficial.txt
        try (BufferedReader reader = new BufferedReader(new FileReader(inputFileName))) {
            String line = reader.readLine();
            if (line != null) {
                n = Integer.parseInt(line.trim());
                System.out.println("Read number: " + n);
            } else {
                System.out.println("No number found in " + inputFileName);
                return;
            }
        } catch (IOException e) {
            System.out.println("Error reading from " + inputFileName + ": " + e.getMessage());
            return;
        }

        // Generate the diamond pattern and write to output.txt
        try (BufferedWriter writer = new BufferedWriter(new FileWriter(outputFileName))) {
            String diamond = generateDiamond(n);
            writer.write(diamond);
            System.out.println("Diamond pattern written to " + outputFileName);
        } catch (IOException e) {
            System.out.println("Error writing to " + outputFileName + ": " + e.getMessage());
        }
    }

    public static String generateDiamond(int n) {
        StringBuilder sb = new StringBuilder();

        // Print the top half of the diamond
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                sb.append(" ");
            }
            for (int j = 0; j < 2 * i + 1; j++) {
                sb.append("*");
            }
            sb.append("\n");
        }

        // Print the bottom half of the diamond
        for (int i = n - 2; i >= 0; i--) {
            for (int j = 0; j < n - i - 1; j++) {
                sb.append(" ");
            }
            for (int j = 0; j < 2 * i + 1; j++) {
                sb.append("*");
            }
            sb.append("\n");
        }

        return sb.toString();
    }
}
